featuretools/feature_discovery/type_defs.py

```diff
@@ -0,0 +1,179 @@
+
+        if df_id:
+            hash_msg.update(df_id.encode("utf-8"))
+
+        if primitive:
+            primitive_name = primitive.name
+            assert isinstance(primitive_name, str)
+            commutative = primitive.commutative
+            hash_msg.update(primitive_name.encode("utf-8"))
+
+            assert (
+                len(base_features) > 0
+            ), "there must be base features if give a primitive"
+            base_columns = base_features
+            if commutative:
+                base_features.sort()
+
+            for c in base_columns:
+                hash_msg.update(c.id.encode("utf-8"))
+
+        else:
+            assert name
+            hash_msg.update(name.encode("utf-8"))
+
+        return hash_msg.hexdigest()
+
+    def __eq__(self, other: Feature):
+        return self.id == other.id
+
+    def __lt__(self, other: Feature):
+        return self.id < other.id
+
+    def __ne__(self, other):
+        return self.id != other.id
+
+    def __hash__(self):
+        return hash(self.id)
+
+    def _generate_hash(self) -> str:
+        return self.hash(
+            name=self.name,
+            primitive=self.primitive,
+            base_features=self.base_features,
+            df_id=self.df_id,
+        )
+
+    def __post_init__(self):
+        self.id = self._generate_hash()
+
+    def to_dict(self):
+        return {
+            "name": self.name,
+            "logical_type": self.logical_type.__name__ if self.logical_type else None,
+            "tags": list(self.tags),
+            "primitive": self.primitive.__name__ if self.primitive else None,
+            "base_features": [x.to_dict() for x in self.base_features],
+            "df_id": self.df_id,
+            "id": self.id,
+        }
+
+    @staticmethod
+    def from_dict(input_dict: Dict) -> Feature:
+        base_features = [Feature.from_dict(x) for x in input_dict["base_features"]]
+
+        primitive = (
+            primitives_map[input_dict["primitive"]] if input_dict["primitive"] else None
+        )
+
+        logical_type = (
+            logical_types_map[input_dict["logical_type"]]
+            if input_dict["logical_type"]
+            else None
+        )
+
+        hydrated_feature = Feature(
+            name=input_dict["name"],
+            logical_type=logical_type,
+            tags=set(input_dict["tags"]),
+            primitive=primitive,
+            base_features=base_features,
+            df_id=input_dict["df_id"],
+        )
+
+        assert hydrated_feature.id == input_dict["id"]
+
+        return hydrated_feature
+
+
+def convert_old_to_new(feature: FeatureBase):
+    base_features = [convert_old_to_new(x) for x in feature.base_features]
+
+    name = feature.get_name()
+    col_schema = feature.column_schema
+
+    logical_type = col_schema.logical_type
+    if logical_type is not None:
+        assert issubclass(type(logical_type), LogicalType)
+        logical_type = type(logical_type)
```

Comment:
```suggestion
    if (logical_type := col_schema.logical_type):
        assert issubclass(type(logical_type), LogicalType)
        logical_type = type(logical_type)
```
I _think_ you can do this(?)